fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/services/ReplacePropertiesServiceImplTest.java

```diff
@@ -80,46 +79,46 @@
 
     private RdfSourceOperationFactory factory;
 
-    private static final String FEDORA_URI = "http://www.example.com/fedora/rest/resource1";
+    @Captor
+    private ArgumentCaptor<UpdateRdfSourceOperation> operationCaptor;
+
     private static final String FEDORA_ID = "info:fedora/resource1";
     private static final String TX_ID = "tx-1234";
     private static final String CONTENT_TYPE = "text/turtle";
 
     private static final String RDF =
-            "@prefix dc: <"  + DC.getURI() + "> ." +
-            "@prefix dcterms: <"  + DCTerms.getURI() + "> ." +
-            "<" + FEDORA_URI + "> dc:title 'fancy title' ;" +
-            "<" + FEDORA_URI + "> dc:title 'another fancy title' ;";
+            "<" + FEDORA_ID + "> <" + DC.getURI() + "title> 'fancy title' .\n" +
+            "<" + FEDORA_ID + "> <" + DC.getURI() + "title> 'another fancy title' .";
 
     private final Node subject = createURI(FEDORA_ID);
 
-    private final Triple triple1 = new Triple(createURI(FEDORA_ID),
-        createURI("http://purl.org/dc/elements/1.1/title"),
-        createLiteral("title one", XSDDatatype.XSDstring));
-    private final Triple triple2 = new Triple(createURI(FEDORA_ID),
-        createURI("http://purl.org/dc/elements/1.1/title"),
-        createLiteral("title two", XSDDatatype.XSDstring));
-
 
     @Before
     public void setup() {
+        factory = new RdfSourceOperationFactoryImpl();
+        setField(service, "factory", factory);
         when(tx.getId()).thenReturn(TX_ID);
         when(psManager.getSession(anyString())).thenReturn(pSession);
         when(resource.getId()).thenReturn(FEDORA_ID);
-        when(factory.updateBuilder(eq(FEDORA_ID))).thenReturn(builder);
-        when(builder.build()).thenReturn(operation);
     }
 
     @Test
-    @Ignore
-    public void testReplaceProperties() throws PersistentStorageException {
-        final DefaultRdfStream rdfStream = new DefaultRdfStream(subject, of(triple1, triple2));
-        when(resource.getTriples()).thenReturn(rdfStream);
-
+    public void testReplaceProperties() throws Exception {
         final Model model = ModelFactory.createDefaultModel();
+        RDFDataMgr.read(model, IOUtils.toInputStream(RDF, "UTF-8"), Lang.NTRIPLES);
 
         service.perform(tx.getId(), resource.getId(), CONTENT_TYPE, model);
-        verify(pSession).persist(operation);
+        verify(pSession).persist(operationCaptor.capture());
+        assertEquals(FEDORA_ID, operationCaptor.getValue().getResourceId());
+        final RdfStream stream = operationCaptor.getValue().getTriples();
+        final Model captureModel = stream.collect(RdfCollectors.toModel());
+
+        assertTrue(captureModel.contains(ResourceFactory.createResource(FEDORA_ID),
+                ResourceFactory.createProperty("http://purl.org/dc/elements/1.1/title"),
```

Comment:
I think you can just say `DC.title` here and not create anything.  Same for the next assert.

Reply:
Oh, nm, I thought this class was using that already and I see it isn't.